lib/src/analyzers/lint_analyzer/rules/rules_list/avoid_banned_imports/utils/config_parser.dart

```diff
@@ -0,0 +1,40 @@
+part of '../avoid_banned_imports_rule.dart';
+
+const _entriesLabel = 'entries';
+const _pathsLabel = 'paths';
+const _denyLabel = 'deny';
+const _messageLabel = 'message';
+
+/// Parser for rule configuration.
+class _ConfigParser {
+  static List<_AvoidBannedImportsConfigEntry> _parseEntryConfig(
+    Map<String, Object> config,
+  ) =>
+      (config[_entriesLabel] as Iterable<Object?>? ?? []).map((entry) {
+        final entryMap = entry as Map<Object?, Object?>;
+
+        return _AvoidBannedImportsConfigEntry(
+          paths: _parseListRegExp(entryMap[_pathsLabel]),
+          deny: _parseListRegExp(entryMap[_denyLabel]),
+          message: entryMap[_messageLabel] as String,
+        );
+      }).toList();
+
+  static List<RegExp> _parseListRegExp(Object? object) =>
+      (object! as List<Object?>)
```

Comment:
Those casts and null assertions look fragile to me. Is it possible to rewrite it with `as` and `whereType`?

Reply:
IMHO fail fast is important. If the user provides wrong types it should immediately fail.

I just mimic `json_serializable`'s auto gen code.

Reply:
Have you tested it with IDE highlights? I'm not sure, that plugin will be recreated on the config change. Could you check that?
For example, enable the rule in other repository referencing local copy, set the wrong config and then set the right one. If the plugin starts showing the errors for this rule - everything is fine

Reply:
But in general I agree with fail fast approach, but I not sure that plugins are 😄

Reply:
I guess we can see it at least in the analysis server's webpage. No bandwidth to try it though.

Reply:
Hm, I actually noticed that saving changes to analysis_options leads to the analyzer restart. I think it might work fine.